### PR TITLE
Modifying the number of leave days calculation part.

### PR DESCRIPTION
**Pull request by meatechsupport**
we don't need to exclude the Holiday list (that comes in between) from the total number of leaves applied.

### CODE REVIEW

**Comment by rmehta** [comment]
@meatechsupport can you create a new field in HR Settings like "Holidays between leaves are counted as leaves" (`include_holidays_in_leaves`) and then base this calculation based on what the setting is?

you can use `frappe.db.get_value("HR Settings", "HR Settings", "include_holidays_in_leaves")` to check the value of the setting.

**Comment by rmehta** [comment]
@meatechsupport do you still want to do this. We can do it as paid development, in 1 hr? https://erpnext.com/pricing/paid-feature-development

**Comment by meatechsupport** [comment]
@rmehta , sorry for the delayed input. i believe the adding the check-box in the "Leave Type" doctype will be more friendly than the  HR settings. Currently i have added Check-box "Include Holiday" in the Leave Type and modified the leave_application.py accordingly.

**Comment by rmehta** [comment]
@meatechsupport Better. You have to also commit the leave_type.json file with the new field.

**Comment by meatechsupport** [comment]
@rmehta committed the leave_type.json.

**Comment by rmehta** [comment]
@meatechsupport thanks, can you change the label to "Include holidays within leaves as leaves" so that it is explicit?

**Comment by meatechsupport** [comment]
@rmehta thanks, the label has been changed accordingly.

File: erpnext/hr/doctype/leave_application/leave_application.py [hunk 1]
```diff
@@ -239,9 +239,14 @@ def get_total_leave_days(leave_app):
 	if not leave_app.half_day:
 		tot_days = date_diff(leave_app.to_date, leave_app.from_date) + 1
 		holidays = leave_app.get_holidays()
+		if frappe.db.get_value("Leave Type", self.leave_type, "include_holiday"):
 		ret = {
-			'total_leave_days' : flt(tot_days)-flt(holidays)
-		}
+			'total_leave_days' : flt(tot_days) - flt(holidays)
+			}
+			else:
```
Comment (nabinhait) [inline erpnext/hr/doctype/leave_application/leave_application.py]: Indentation is not correct

File: erpnext/hr/doctype/leave_application/leave_application.py [hunk 2]
```diff
@@ -239,9 +239,14 @@ def get_total_leave_days(leave_app):
 	if not leave_app.half_day:
 		tot_days = date_diff(leave_app.to_date, leave_app.from_date) + 1
 		holidays = leave_app.get_holidays()
+		if frappe.db.get_value("Leave Type", self.leave_type, "include_holiday"):
```
Comment (nabinhait) [inline erpnext/hr/doctype/leave_application/leave_application.py]: `holidays = leave_app.get_holidays()` can be inside if condition

**Comment by meatechsupport** [comment]
@nabinhait, i hope the changes are good enough to go. :)

**Comment by rmehta** [comment]
@meatechsupport can you write a simple test case? Just copy this one and add your condition in a new Leave Type:

https://github.com/frappe/erpnext/blob/develop/erpnext/hr/doctype/salary_slip/test_salary_slip.py#L28-L29

That will make is future-proof

**Comment by meatechsupport** [comment]
hi @rmehta, test case in the sense!! can please explain?

**Comment by rmehta** [comment]
Did you see the link? A test case ensures that your code runs as it is designed to without anyone checking and if a future update breaks it, then it will throw a failed test.

**Comment by meatechsupport** [comment]
hi @rmehta, have copied it to the test file.

**Comment by meatechsupport** [comment]
@rmehta , any update

**Comment by meatechsupport** [comment]
@rmehta i have tested it on the virtual box image and it was working fine. so can pleas tell what need to be done. sorry still didn't get clear picture about adding test case here. :(

**Comment by rmehta** [comment]
@meatechsupport did you read the other test cases in the file?

Broad introduction to unit testing

https://en.wikipedia.org/wiki/Unit_testing

**Comment by meatechsupport** [comment]
@rmehta  please let me know if further changes required. :)

**Comment by rmehta** [comment]
Nice! Now cleanup.

1. Reset the value of include_holidays in leave_type after the test has passed, because the tests are run in random order and you should leave any changes as they were before the test.
2. Squash your 7 commits into one commit so that it looks cleaner (Google how to do this)
3. Rebase your branch

**Comment by meatechsupport** [comment]
@rmehta how can we know if the status of the test, if it passed or failed?
should i wait till the test finish to squashing and rebasing

**Comment by rmehta** [comment]
No just reset the leave_type setting using 

```
frappe.db.set_value("Leave Type", self.leave_type, "include_holiday", [original setting])
```

and squash and rebase.

To test locally, check this:

```
bench --site [your site] run-tests --doctype "Leave Application"
```

**Comment by meatechsupport** [comment]
@rmehta i really appreciate your help. 

could you please specify which file to edit, leave_type.py or leave_type.json?

**Comment by rmehta** [comment]
Your test case. At the end of your test, reset the leave type property as it may affect other tests.

**Comment by meatechsupport** [comment]
@rmehta squashed and rebased.

**Comment by meatechsupport** [comment]
@rmehta even if i remove the Test Case it fails in testing other function. please let me know if i ignore and add the test case again. waiting  for your response.

**Comment by rmehta** [comment]
@meatechsupport @nabinhait  will merge your PR later today.

**Comment by nabinhait** [comment]
Merged. Made some related changes as well, https://github.com/frappe/erpnext/commit/b24d2efc4b99239f935b87a685e46cb5da2497c6.

**Pull request merged**